pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/SegmentFetcherAndLoader.java

```diff
@@ -201,11 +201,10 @@ public void addOrReplaceOfflineSegment(String tableName, String segmentId, boole
             }
           }
         }
-        if (_segmentLoadMaxRetryCount <= retryCount) {
-          String msg =
-              "Failed to download segment " + segmentId + " (table " + tableName + " after " + retryCount + " retries";
-          LOGGER.error(msg);
-          throw new RuntimeException(msg);
+        if (retryCount == maxRetryCount) {
```

Comment:
Will address that in separate PR. This one is just to fix the problem where reload failure does not log any error.